pkg/gardenlet/controller/managedseed/managedseed_reconciler.go

```diff
@@ -0,0 +1,280 @@
+		managedSeedLogger.Infof("Waiting for shoot %s to be reconciled before registering it as seed", kutil.ObjectName(shoot))
+
+		// Update status to Pending
+		if updateErr := updateStatusPending(r.ctx, gardenClient.DirectClient(), managedSeed, "Reconciliation pending"); updateErr != nil {
+			managedSeedLogger.Errorf("Could not update ManagedSeed status for pending creation or update: %+v", updateErr)
+			return reconcile.Result{}, updateErr
+		}
+
+		// Return success result with requeue after 10 seconds
+		return reconcile.Result{RequeueAfter: 10 * time.Second}, nil
+	}
+
+	// Get seed client
+	seedClient, err := r.clientMap.GetClient(r.ctx, keys.ForSeedWithName(*shoot.Spec.SeedName))
+	if err != nil {
+		return reconcile.Result{}, fmt.Errorf("could not get seed client: %w", err)
+	}
+
+	// Get shoot client
+	shootClient, err := r.clientMap.GetClient(r.ctx, keys.ForShoot(shoot))
+	if err != nil {
+		return reconcile.Result{}, fmt.Errorf("could not get shoot client: %w", err)
+	}
+
+	// Create actuator
+	a := newActuator(gardenClient, seedClient, shootClient, r.config, r.imageVector, r.logger)
+
+	// Update status to Processing
+	if updateErr := updateStatusProcessing(r.ctx, gardenClient.DirectClient(), managedSeed, "Reconciling"); updateErr != nil {
+		managedSeedLogger.Errorf("Could not update ManagedSeed status before creation or update: %+v", updateErr)
+		return reconcile.Result{}, updateErr
+	}
+
+	// Reconcile creation or update
+	managedSeedLogger.Debug("Reconciling ManagedSeed creation or update")
+	if err := a.CreateOrUpdate(r.ctx, managedSeed, shoot); err != nil {
+		managedSeedLogger.Errorf("Could not reconcile ManagedSeed creation or update: %+v", err)
+
+		// Record an event
+		lastError := &gardencorev1beta1.LastError{
+			Codes:       gardencorev1beta1helper.ExtractErrorCodes(err),
+			Description: "Could not reconcile creation or update: " + err.Error(),
+		}
+		r.recorder.Eventf(managedSeed, corev1.EventTypeWarning, gardencorev1beta1.EventReconcileError, "%s", lastError.Description)
+
+		// Update status to Error
+		if updateErr := updateStatusError(r.ctx, gardenClient.DirectClient(), managedSeed, lastError.Description, lastError); updateErr != nil {
+			managedSeedLogger.Errorf("Could not update ManagedSeed status after creation or update error: %+v", updateErr)
+			return reconcile.Result{}, updateErr
+		}
+
+		// Return error result
+		return reconcile.Result{}, errors.New(lastError.Description)
+	}
+	managedSeedLogger.Debug("Successfully reconciled ManagedSeed creation or update")
+
+	// Update status to Succeeded
+	if updateErr := updateStatusSucceeded(r.ctx, gardenClient.DirectClient(), managedSeed, "Reconciled successfully"); updateErr != nil {
+		managedSeedLogger.Errorf("Could not update ManagedSeed status after creation or update success: %+v", updateErr)
+		return reconcile.Result{}, updateErr
+	}
+
+	// Return success result
+	return reconcile.Result{}, nil
```

Comment:
Should we have a sync period for this controller, after which a `ManagedSeed` is automatically reconciled again?

Reply:
I don't know, should we? What is the criteria to opt for one? Sorry, I am a bit unsure of the implications, do you have something specific in mind?

Reply:
We decided not to have a sync period for now.